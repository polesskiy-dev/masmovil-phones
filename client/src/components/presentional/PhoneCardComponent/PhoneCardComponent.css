.phone-card {
    display: flex;
    flex-direction: column;
    justify-content: center;

    background-color: var(--phone-card-bg-color);

    border: 1px solid rgba(0,0,0,.125);
    border-radius: .25rem;
}

.phone-card .phone-card__image {
    max-height: 256px;
    object-fit: contain;

    padding: 16px;
}

.phone-card .phone-card__description {
    display: flex;
    flex-direction: column;
    justify-content: space-between;

    padding: 16px;
}

.phone-card .phone-card__name {
    color: var(--phone-card-color);

    font-size: 1.2em;
}

.phone-card  .phone-card__details {
    color: var(--phone-card-color);
}