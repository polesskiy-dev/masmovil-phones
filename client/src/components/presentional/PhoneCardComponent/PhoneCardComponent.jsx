import React from 'react';
import PropTypes from 'prop-types';

import './PhoneCardComponent.css';

import DetailsButtonComponent from '../DetailsButtonComponent/DetailsButtonComponent';

const PhoneCardComponent = ({ DeviceName, img, type, os }) => (
  <article className="phone-card">
    <img className="phone-card__image" src={img} alt={DeviceName} />
    <div className="phone-card__description">
      <div className="phone-card__text-description">
        <p className="phone-card__name">{DeviceName}</p>
        <ul className="phone-card__details">
          <li>{type}</li>
          <li>{os}</li>
        </ul>
      </div>
      <DetailsButtonComponent />
    </div>
  </article>
);

PhoneCardComponent.propTypes = {
  DeviceName: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  os: PropTypes.string.isRequired,
};

export default PhoneCardComponent;
