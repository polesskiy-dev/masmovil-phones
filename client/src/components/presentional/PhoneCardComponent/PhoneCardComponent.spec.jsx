import React from 'react';
import renderer from 'react-test-renderer';
import _ from 'lodash/fp';

import withDefaultIntl from '../../../helpers/with-default-intl';
import PhoneCardComponent from './PhoneCardComponent';

it('<PhoneDetailComponent/ > renders correctly with all required fields', () => {
  const tree = renderer
    .create(
      withDefaultIntl(
        <PhoneCardComponent
          type=""
          DeviceName=""
          img=""
          os=""
          showPhoneDetails={_.noop}
        />
      )
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
