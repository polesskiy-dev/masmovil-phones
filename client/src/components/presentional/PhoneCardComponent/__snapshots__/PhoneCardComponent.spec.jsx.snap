// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhoneDetailComponent/ > renders correctly with all required fields 1`] = `
<article
  className="phone-card"
>
  <img
    alt=""
    className="phone-card__image"
    src=""
  />
  <div
    className="phone-card__description"
  >
    <div
      className="phone-card__text-description"
    >
      <p
        className="phone-card__name"
      >
        
      </p>
      <ul
        className="phone-card__details"
      >
        <li>
          
        </li>
        <li>
          
        </li>
      </ul>
    </div>
    <button
      className="details-button"
      onClick={[Function]}
    >
      <span>
        Details
      </span>
    </button>
  </div>
</article>
`;
